docs/source/data/index.md

```diff
@@ -1,5 +1,5 @@
 
-# The Kedro Data Catalog
+# Data Catalog
```

Comment:
```suggestion
# DataCatalog
```
nit, since the heading below is `KedroDataCatalog`

Reply:
I kept **Data Catalog** title for the index page and **Kedro Data Catalog** for the page related to the new catalog as quoted class names do not look nice in the main headers IMO. I also quoted `KedroDataCatalog` in the sub-header on the index page.